File: 0058-length-of-last-word/0058-length-of-last-word.cpp
class Solution {
public:
    int lengthOfLastWord(string s) {
        int start = -1;
        int end = -1;
        for(int i=s.size()-1;i>=0;i--)
        {
            if(s[i]!=' ')
            {
                end = i;
                break;
            }
        }
        for(int i=end;i>=0;i--)
        {
            if(s[i]==' ')
            {
                start = i;
                break;
            }
        }
        //cout<<start<<" "<<end<<endl;
        return end - start;
    }
};

File: 0058-length-of-last-word/0058-length-of-last-word.java
class Solution {
    public int lengthOfLastWord(String s) {
        
        int i = s.length()-1;
        
        while(s.charAt(i) == ' ')
        {
            i--;
        }
        
        int end = i;
        
        while(i!=-1 && s.charAt(i) != ' '){
            i--;
        }
        
        int start = i;
        
        return end - start;
    }
}